### ISSUE TITLE
type Amp\Socket\EncryptableSocket or Amp\Socket\Socket

### ISSUE THREAD

**Issue by JadRho** [amphp/http-client#313]
Part of composer.json
```
"minimum-stability": "dev",
  "require": {
      "amphp/amp": "3.x-dev",
      "revolt/event-loop": "^1.0@dev",
      "amphp/http-client": "v5.x-dev"
  }
```

Php test file
```
<?php declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';

use Amp\Http\Client\Client;
use Amp\Http\Client\HttpClientBuilder;
use Amp\Http\Client\Request;

$client = HttpClientBuilder::buildDefault();

$response = $client->request(new Request("https://httpbin.org/get"));

var_dump($response->getStatus());
var_dump($response->getHeaders());
var_dump($response->getBody()->buffer());
```

Results in error
```
PHP Fatal error:  Uncaught TypeError: Amp\Http\Client\Connection\Http2Connection::__construct(): Argument #1 ($socket) must be of type Amp\Socket\EncryptableSocket, Amp\Socket\ResourceSocket given, called in /demo/path/vendor/amphp/http-client/src/Connection/DefaultConnectionFactory.php on line 184 and defined in /demo/path/vendor/amphp/http-client/src/Connection/Http2Connection.php:30
Stack trace:
#0 /demo/path/vendor/amphp/http-client/src/Connection/DefaultConnectionFactory.php(184): Amp\Http\Client\Connection\Http2Connection->__construct()
#1 /demo/path/vendor/amphp/amp/src/functions.php(34): Amp\Http\Client\Connection\DefaultConnectionFactory->create()
#2 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(406): Amp\{closure}()
#3 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(542): Revolt\EventLoop\Internal\AbstractDriver->invokeMicrotasks()
#4 [internal function]: Revolt\EventLoop\Internal\AbstractDriver->Revolt\EventLoop\Internal\{closure}()
#5 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(406): Fiber->resume()
#6 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(596): Revolt\EventLoop\Internal\AbstractDriver->invokeMicrotasks()
#7 [internal function]: Revolt\EventLoop\Internal\AbstractDriver->Revolt\EventLoop\Internal\{closure}()
#8 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(478): Fiber->resume()
#9 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(533): Revolt\EventLoop\Internal\AbstractDriver->invokeCallbacks()
#10 [internal function]: Revolt\EventLoop\Internal\AbstractDriver->Revolt\EventLoop\Internal\{closure}()
#11 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/AbstractDriver.php(86): Fiber->start()
#12 /demo/path/vendor/revolt/event-loop/src/EventLoop/Internal/DriverSuspension.php(97): Revolt\EventLoop\Internal\AbstractDriver->Revolt\EventLoop\Internal\{closure}()
#13 /demo/path/vendor/amphp/amp/src/Internal/FutureIterator.php(133): Revolt\EventLoop\Internal\DriverSuspension->suspend()
#14 /demo/path/vendor/amphp/amp/src/Future.php(59): Amp\Internal\FutureIterator->consume()
#15 /demo/path/vendor/amphp/amp/src/Future/functions.php(26): Amp\Future::iterate()
#16 /demo/path/vendor/amphp/http-client/src/Connection/ConnectionLimitingPool.php(248): Amp\Future\awaitFirst()
#17 /demo/path/vendor/amphp/http-client/src/Connection/ConnectionLimitingPool.php(109): Amp\Http\Client\Connection\ConnectionLimitingPool->getStreamFor()
#18 /demo/path/vendor/amphp/http-client/src/Connection/UnlimitedConnectionPool.php(42): Amp\Http\Client\Connection\ConnectionLimitingPool->getStream()
#19 /demo/path/vendor/amphp/http-client/src/PooledHttpClient.php(31): Amp\Http\Client\Connection\UnlimitedConnectionPool->getStream()
#20 /demo/path/vendor/amphp/http-client/src/Interceptor/RetryRequests.php(36): Amp\Http\Client\PooledHttpClient->request()
#21 /demo/path/vendor/amphp/http-client/src/InterceptedHttpClient.php(30): Amp\Http\Client\Interceptor\RetryRequests->request()
#22 /demo/path/vendor/amphp/http-client/src/Interceptor/FollowRedirects.php(129): Amp\Http\Client\InterceptedHttpClient->request()
#23 /demo/path/vendor/amphp/http-client/src/InterceptedHttpClient.php(30): Amp\Http\Client\Interceptor\FollowRedirects->request()
#24 /demo/path/vendor/amphp/http-client/src/Interceptor/ModifyRequest.php(52): Amp\Http\Client\InterceptedHttpClient->request()
#25 /demo/path/vendor/amphp/http-client/src/InterceptedHttpClient.php(30): Amp\Http\Client\Interceptor\ModifyRequest->request()
#26 /demo/path/vendor/amphp/http-client/src/HttpClient.php(26): Amp\Http\Client\InterceptedHttpClient->request()
#27 /demo/path/anima.php(14): Amp\Http\Client\HttpClient->request()
#28 {main}
  thrown in /demo/path/vendor/amphp/http-client/src/Connection/Http2Connection.php on line 30
```
Since the change in amphp/socket 2.0.0
```
Moved methods from the EncryptableSocket interface to Socket and removed EncryptableSocket
```
Changing **EncryptableSocket** to **Socket** in
**amphp/http-client/src/Connection/Http2Connection.php:30** and 
**amphp/http-client/src/Connection/Internal/Http2ConnectionProcessor.php:100**
Should be the correct solution?

**Comment by JadRho** [comment]
Fixed in commit [2043047f4554e3b0b090423f81f743c5f746b854](https://github.com/amphp/http-client/commit/2043047f4554e3b0b090423f81f743c5f746b854)
Thank you @trowski

**Comment by JadRho** [comment]
Fixed in commit [2043047f4554e3b0b090423f81f743c5f746b854](https://github.com/amphp/http-client/commit/2043047f4554e3b0b090423f81f743c5f746b854)
Thank you @trowski
